src/olympia/addons/fields.py

```diff
@@ -240,27 +241,27 @@ def to_internal_value(self, data):
             if isinstance(data, dict):
                 version = self.parent.instance
                 existing = version.compatible_apps if version else {}
-                qs = AppVersion.objects
                 internal = {}
                 for app_name, min_max in data.items():
                     app = amo.APPS[app_name]
-                    apps_versions = existing.get(
-                        app, ApplicationsVersions(application=app.id)
-                    )
+                    # we need to copy() to avoid changing the instance before save
+                    apps_versions = (
+                        (ex := existing.get(app)) and copy.copy(ex)
+                    ) or ApplicationsVersions(application=app.id)
```

Comment:
this was kind of a bug in the existing code, but harmless - `to_internal_value` is supposed to only return the new value, and isn't supposed to change anything on the version instance, but it was, subtly.  It was harmless because it was only changing a cached property and the instance wasn't saved in any case.

Reply:
It only became a problem when I needed to work out what had changed compared to the original instance for logging.